pkgs/development/python-modules/asgi-lifespan/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+
+# build-system
+, setuptools-scm
+
+# dependencies
+, twine
+, wheel
+, attrs
+, autoflake
+, black
+, mypy
+, pytest
+, ruff
+, starlette
+, trio
+, httpx
+, setuptools
+
+# tests
+ }:
+
+buildPythonPackage rec {
+  pname = "asgi-lifespan";
+  version = "2.1.0";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-Xi7/rwv+OYKc8tZOfsxHx9htZ2plmfevujeMMfXjowg=";
+  };
+
+  postPatch = ''
+    # don't test bash builtins
+    #rm testing/test_argcomplete.py
+  '';
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
+    setuptools
+    starlette
+  ];
+
+  nativeCheckInputs = [
+  ];
+
+  doCheck = false;
```

Comment:
If tests are disabled then a comment should be added about the why.

Please add `pythonImportsCheck` as additional safe-guard.